gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/ChannelPool.java

```diff
@@ -415,10 +415,33 @@ private Entry getEntry(int affinity) {
     return localEntries.get(index);
   }
 
+  private static class RetainedEntry {
```

Comment:
Can this extends `Entry` so all the existing references to `Entry` can stay as they are?

Reply:
no, Entry represents a shared count, while this represents a single lease from the shared count

Reply:
I see, can we add some comments on this class?

Reply:
done